### PR TITLE
Initial dev setup

### PR DESCRIPTION
**Pull request by DanielChicot**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ancillary-images: java-image python-image dks-image s3-init-image  ## Build base images
+
+build-jar: ## Build the jar.
+	./gradlew clean build
+
+dist: ## Assemble distribution files in build/dist.
+	./gradlew assembleDist
+
+.PHONY: build-all
+build-all: build-jar build-images ## Build the jar file and the images.
```
Comment (dan-hill2802) [inline Makefile]: this doesn't match the job name: `build-image` vs `build-images`

File: docker/dks/application.properties [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+server.port=8443
+server.ssl.client-auth=need
+server.ssl.key-alias=cid
+server.ssl.key-store-password=changeit
+server.ssl.key-store-type=JKS
+server.ssl.key-store=dks-keystore.jks
+server.ssl.trust-alias=cid
+server.ssl.trust-store-type=JKS
+server.ssl.trust-store-password=changeit
+server.ssl.trust-store=dks-truststore.jks
```
Comment (dan-hill2802) [inline docker/dks/application.properties]: new line at end

File: src/main/kotlin/app/batch/EncryptionProcessor.kt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+    override fun process(item: String): String? {
+        logger.info("httpClientProvider: '$httpClientProvider'.")
+        return item.toUpperCase()
+    }
+
+    companion object {
+        val logger: Logger = LoggerFactory.getLogger(EncryptionProcessor::class.toString())
+    }
+}
```
Comment (dan-hill2802) [inline src/main/kotlin/app/batch/EncryptionProcessor.kt]: new line

File: src/main/kotlin/app/batch/DecryptionProcessor.kt [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+    override fun process(item: String): String? {
+        logger.info("httpClientProvider: '$httpClientProvider'.")
+        return "{ \"item\": \"$item\" }"
+    }
+
+    companion object {
+        val logger: Logger = LoggerFactory.getLogger(DecryptionProcessor::class.toString())
+    }
+}
```
Comment (dan-hill2802) [inline src/main/kotlin/app/batch/DecryptionProcessor.kt]: new line

File: src/main/kotlin/app/batch/HBaseWriter.kt [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        items.forEach {
+            logger.info("it: '$it'.")
+        }
+    }
+
+    companion object {
+        val logger: Logger = LoggerFactory.getLogger(HBaseWriter::class.toString())
+    }
+
+}
```
Comment (dan-hill2802) [inline src/main/kotlin/app/batch/HBaseWriter.kt]: new line

File: src/main/kotlin/app/batch/S3Reader.kt [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        else {
+            null
+        }
+    }
+    companion object {
+        val logger: Logger = LoggerFactory.getLogger(HBaseWriter::class.toString())
+    }
+
+    private var count = 0
+}
```
Comment (dan-hill2802) [inline src/main/kotlin/app/batch/S3Reader.kt]: new line

File: src/main/kotlin/app/configuration/S3Configuration.kt [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+    @Value("\${s3.service.endpoint}")
+    private lateinit var serviceEndpoint: String
+
+    @Value("\${aws.access.key}")
+    private lateinit var awsAccessKey: String
+
+    @Value("\${aws.secret.key}")
+    private lateinit var awsSecretKey: String
+}
```
Comment (dan-hill2802) [inline src/main/kotlin/app/configuration/S3Configuration.kt]: new line

File: src/main/kotlin/app/configuration/SecureHttpClientProvider.kt [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+    @Value("\${identity.key.password}")
+    private lateinit var identityKeyPassword: String
+
+    @Value("\${trust.keystore}")
+    private lateinit var trustStore: String
+
+    @Value("\${trust.store.password}")
+    private lateinit var trustStorePassword: String
+}
```
Comment (dan-hill2802) [inline src/main/kotlin/app/configuration/SecureHttpClientProvider.kt]: new line

File: src/main/resources/logback.xml [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    parameter store. The following config silences the http library.
+    -->
+    <logger name="org.apache" level="ERROR" additivity="true">
+        <appender-ref ref="CONSOLE" />
+    </logger>
+
+    <root level="INFO">
+        <appender-ref ref="CONSOLE"/>
+    </root>
+</configuration>
```
Comment (dan-hill2802) [inline src/main/resources/logback.xml]: new line

**Comment by dan-hill2802** [comment]
CircleCI looks good, I've skimmed over the Java/Gradle/Kotlin stuff, looking for typos

File: src/main/kotlin/app/batch/EncryptionProcessor.kt (hunk 3 above)
Comment (DanielChicot) [reply src/main/kotlin/app/batch/EncryptionProcessor.kt]: Fixed.

File: src/main/kotlin/app/batch/HBaseWriter.kt (hunk 5 above)
Comment (DanielChicot) [reply src/main/kotlin/app/batch/HBaseWriter.kt]: Added.

File: src/main/kotlin/app/batch/S3Reader.kt (hunk 6 above)
Comment (DanielChicot) [reply src/main/kotlin/app/batch/S3Reader.kt]: Appended.

File: src/main/kotlin/app/configuration/S3Configuration.kt (hunk 7 above)
Comment (DanielChicot) [reply src/main/kotlin/app/configuration/S3Configuration.kt]: There now.

File: src/main/kotlin/app/configuration/SecureHttpClientProvider.kt (hunk 8 above)
Comment (DanielChicot) [reply src/main/kotlin/app/configuration/SecureHttpClientProvider.kt]: Fixed.

File: src/main/kotlin/app/batch/DecryptionProcessor.kt (hunk 4 above)
Comment (DanielChicot) [reply src/main/kotlin/app/batch/DecryptionProcessor.kt]: Done.

File: docker/dks/application.properties (hunk 2 above)
Comment (DanielChicot) [reply docker/dks/application.properties]: Done.

File: src/main/resources/logback.xml (hunk 9 above)
Comment (DanielChicot) [reply src/main/resources/logback.xml]: Done.

File: src/main/kotlin/app/batch/HBaseWriter.kt [hunk 10]
```diff
@@ -0,0 +1,22 @@
+package app.batch
```
Comment (UdayChokkam) [inline src/main/kotlin/app/batch/HBaseWriter.kt:1]: Do we need HBase writer in this project?

Comment (mark-r-m) [reply src/main/kotlin/app/batch/HBaseWriter.kt:1]: Yes afaik, so it can write the data into HBase?

**Review by mark-r-m — APPROVED** [review]

File: src/main/kotlin/app/batch/HBaseWriter.kt (hunk 10 above)
Comment (DanielChicot) [reply src/main/kotlin/app/batch/HBaseWriter.kt:1]: Yes, it writes to hbase.

**Review by UdayChokkam — APPROVED** [review]

**Pull request merged**